core/templates/dev/head/pages/query/Query.js

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Data and controllers for the Oppia admin page.
+ */
+
+oppia.controller('Query', [
+  '$scope', '$http', function($scope, $http) {
+    _QUERY_DATA_URL = '/querydatahandler';
+
+    var _validate = function(data) {
+      return !isNaN(data) || data === null;
+    };
+    $scope.recentQueryOffset = 0;
+    var _baseOffset = 10;
+
+    $scope.submitQuery = function() {
+      if (_validate($scope.has_not_logged_in_for_n_days) &&
+          _validate($scope.inactive_in_last_n_days) &&
+          _validate($scope.created_at_least_n_exps) &&
+          _validate($scope.created_fewer_than_n_exps) &&
+          _validate($scope.edited_at_least_n_exps) &&
+          _validate($scope.edited_fewer_than_n_exps)) {
+        data = {
+          has_not_logged_in_for_n_days: $scope.has_not_logged_in_for_n_days,
```

Comment:
Sort of like this, I was thinking: https://select2.github.io/examples.html#tags -- but I'm not sure how to handle the additional number parameter.

With reset, will it be clear when a field is empty versus when it has a value of zero? If so, I think that might be OK, since this page has a small audience; we can make it doubly-clear by putting a check mark on the right of the field if it's non-empty. Might want to style the form nicely, though, using ,form-control or similar -- see e.g. the input fields in the editor settings tab.